cms/djangoapps/import_from_modulestore/models.py

```diff
@@ -0,0 +1,172 @@
+
+from .data import ImportStatus
+
+User = get_user_model()
+
+
+class Import(TimeStampedModel):
+    """
+    Represents the action of a user importing a modulestore-based course or legacy
+    library into a learning-core based learning package (today, that is always a content library).
+    """
+
+    uuid = models.UUIDField(default=uuid_tools.uuid4, editable=False, unique=True, db_index=True)
+    status = models.CharField(max_length=100, choices=ImportStatus.choices, default=ImportStatus.PENDING)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    source_key = LearningContextKeyField(help_text=_('The modulestore course'), max_length=255, db_index=True)
+    target = models.ForeignKey(LearningPackage, models.SET_NULL, null=True)
+
+    def __str__(self):
+        return f'{self.source_key} - {self.target}'
+
+    def ready(self) -> None:
+        """
+        Set import status to ready.
+        """
+        self.status = ImportStatus.READY
+        self.save()
+
+    def imported(self) -> None:
+        """
+        Set import status to imported and clean related staged content.
+        """
+        self.status = ImportStatus.IMPORTED
+        self.save()
+        self.clean_related_staged_content()
+
+    def cancel(self) -> None:
+        """
+        Cancel import action and delete related staged content.
+        """
+        self.status = ImportStatus.CANCELED
+        self.save()
+        self.clean_related_staged_content()
+
+    def error(self) -> None:
+        """
+        Set import status to error and clean related staged
+        """
+        self.status = ImportStatus.ERROR
+        self.save()
+        self.clean_related_staged_content()
+
+    def clean_related_staged_content(self) -> None:
+        """
+        Clean related staged content.
+        """
+        for staged_content_for_import in self.staged_content_for_import.all():
+            staged_content_for_import.staged_content.delete()
+
+    class Meta:
+        verbose_name = _('Import from modulestore')
+        verbose_name_plural = _('Imports from modulestore')
+
+    @classmethod
+    def get_by_uuid(cls, import_uuid: str) -> Self | None:
+        """
+        Get an import task by its ID.
+        """
+        return cls.objects.filter(uuid=import_uuid).first()
+
+    @classmethod
+    def get_ready_by_uuid(cls, import_uuid: str) -> Self | None:
+        """
+        Get an import task by its UUID.
+        """
+        return cls.objects.filter(uuid=import_uuid, status=ImportStatus.READY).first()
+
+    def get_staged_content_by_block_usage_id(self, block_usage_id: str) -> Optional["StagedContent"]:
+        """
+        Get staged content by block usage ID.
+        """
+        staged_content_for_import = self.staged_content_for_import.filter(
+            staged_content__tags__icontains=block_usage_id
+        ).first()
+        return getattr(staged_content_for_import, 'staged_content', None)
```

Comment:
Generally speaking, avoid using `getattr`, unless you really really need it. getattr is dangerous, because if you misspell the attribute name, you will never find out--you will just quietly get `None` as your result! It also makes it hard for pylint and mypy to analyze your code.

If you need to handle None, then just handle None:
```suggestion
        return staged_content_for_import.staged_content if staged_content_for_import else None
```

Reply:
Updated.